server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -86,7 +86,11 @@ public TransportGetSnapshotsAction(
             GetSnapshotsRequest::new,
             indexNameExpressionResolver,
             GetSnapshotsResponse::new,
-            ThreadPool.Names.SAME
+            ThreadPool.Names.MANAGEMENT // Execute this on the management pool because creating the response can become fairly expensive
```

Comment:
I can see this being a little controversial given that we generate the response on the meta pool if verbose=true. But in those cases I'd argue that we will probably be bound by the IO before the compute such that we never go 100% CPU on all pool threads simultaneously.